helix-term/src/ui/mod.rs

```diff
@@ -506,18 +507,30 @@ pub mod completers {
             }) // TODO: unwrap or skip
             .filter(|path| !path.is_empty());
 
+        let directory_color = editor.theme.get("ui.text.directory");
+
+        let style_from_file = |file: Cow<'static, str>| {
+            if file.ends_with(std::path::MAIN_SEPARATOR) {
```

Comment:
Instead of checking the end of the string we can actually determine if these files are directories or without the filesystem-access-per-file (mentioned earlier in review) by uncommenting from above:

```rust
let is_dir = entry.file_type().is_some_and(|entry| entry.is_dir());
```

(`DirEntry` already stores the file type and `file_type` just accesses that info, no extra fs access.)

This should fix the colors when using something like `:cd` that uses the directory completer.

There's a bit of overlap between this and #11285 so I would suggest adding a struct near the top of this file

```rust
struct Utf8PathBuf {
    path: String,
    is_dir: bool,
}
```

And the `filter_map` in `filename_impl` can return that type. Then to make it play well with `fuzzy_match` helper we can implement `AsRef<str>` for that struct:

```rust
impl AsRef<str> for Utf8PathBuf {
    fn as_ref(&str) -> &str {
        &self.path
    }
}
```

Then later #11285 can reuse this type instead of the `(PathBuf, bool)` it uses currently.

Reply:
thanks, added in https://github.com/helix-editor/helix/pull/12205/commits/b45d3ebb05cb022529cb97a2d2c13defcf234348